app/src/main/java/org/mozilla/fenix/shortcut/PwaOnboardingDialogFragment.kt

```diff
@@ -38,7 +40,10 @@ class PwaOnboardingDialogFragment : DialogFragment() {
         add_button.setOnClickListener {
             viewLifecycleOwner.lifecycleScope.launch {
                 components.useCases.webAppUseCases.addToHomescreen()
-            }.invokeOnCompletion { dismiss() }
+                requireContext().metrics.track(Event.ProgressiveWebAppAddToHomescreenTap)
```

Comment:
No need to throw an error as we already have a components reference.
```suggestion
                components.analytics.metrics.track(Event.ProgressiveWebAppAddToHomescreenTap)
```